lib/lints/avoid_late_keyword/avoid_late_keyword_rule.dart

```diff
@@ -31,9 +34,28 @@ class AvoidLateKeywordRule extends SolidLintRule {
     CustomLintContext context,
   ) {
     context.registry.addVariableDeclaration((node) {
-      if (node.declaredElement?.isLate ?? false) {
+      if (_shouldLint(node)) {
         reporter.reportErrorForNode(code, node);
       }
     });
   }
+
+  bool _shouldLint(VariableDeclaration node) {
+    final isLateDeclaration = node.declaredElement?.isLate ?? false;
+    if (!isLateDeclaration) return false;
+
+    final ignoredTypes = _hasIgnoredType(node);
+    if (ignoredTypes) return false;
+
+    final allowInitialized = config.parameters.allowInitialized;
+    if (!allowInitialized) return true;
+
+    final hasInitializer = node.initializer != null;
+    return !hasInitializer;
+  }
```

Comment:
So we do not honor `allowInitialized` config if there are no ignored types? Maybe we need to combine them somehow?

Reply:
@maxxlab disregard this -- i misinterpreted the logic flow.